src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -3183,8 +3316,9 @@ public virtual bool CoercesTo(DType typeDest, out bool isSafe, out DType coercio
                 case DKind.Boolean:
                     isSafe = Kind != DKind.String;
                     doesCoerce = Kind == DKind.String ||
-                                 Number.Accepts(this) ||
-                                 Decimal.Accepts(this) ||
+                                 (usePowerFxV1CompatibilityRules && Currency.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true)) ||
```

Comment:
>Currency.Accepts(this, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: true) [](http://example.com/codeflow?start=68&length=106)

Why are we adding this?

Reply:
Currency was accepted by number, so it could be converted to boolean. Now we don't have a currency/number accepts relationship anymore, so we need to add a new coercion node between the two types